projects/jupyter-server-ydoc/jupyter_server_ydoc/handlers.py

```diff
@@ -28,13 +29,17 @@
     LogLevel,
     MessageType,
     decode_file_path,
+    encode_file_path,
     room_id_from_encoded_path,
 )
-from .websocketserver import JupyterWebsocketServer
+from .websocketserver import JupyterWebsocketServer, RoomNotFound
 
 YFILE = YDOCS["file"]
 
+
 SERVER_SESSION = str(uuid.uuid4())
+UNDO_MANAGERS = {}
+FORK_DOCUMENTS = {}
```

Comment:
I don't think we need to keep track of fork documents and undo managers in `FORK_DOCUMENTS` and `UNDO_MANAGERS`. They can both be accessed from the room: `yroom.ydoc` and `yroom.ydoc.undo_manager`.

Reply:
I think the `yroom.ydoc` doesn't have an `undo_manager` but this does https://github.com/jupyter-server/jupyter_ydoc/blob/main/jupyter_ydoc/ybasedoc.py#L10 right ? so to access the `undo_manager` , i do this for example ` fork_document = ydoc_factory(fork_ydoc) `

Reply:
> I think the `yroom.ydoc` doesn't have an `undo_manager` but this does https://github.com/jupyter-server/jupyter_ydoc/blob/main/jupyter_ydoc/ybasedoc.py#L10 right ? so to access the `undo_manager` , i do this for example `fork_document = ydoc_factory(fork_ydoc)`

I could remove UNDO_MANAGERS but I think I need to keep FORK_DOCUMENTS so i could access the undo manager

Reply:
Good point, yes you can at least remove `UNDO_MANAGERS`, or have a subclass of `YRoom` for your `fork_room` where you add a property for the fork document (that way you can get rid of `FORK_DOCUMENTS` too).